test/lightning/collections_test.exs

```diff
@@ -330,6 +353,19 @@ defmodule Lightning.CollectionsTest do
       assert %{value: "value5", updated_at: ^updated_at5} =
                Repo.get_by(Item, key: "key5")
     end
+
+    test "raises Postgrex error when an item is bigger than allowed max len" do
+      collection = insert(:collection)
+
+      items = [
+        %{"key" => "key1", "value" => "adsf"},
+        %{"key" => "key1", "value" => String.duplicate("a", 1_000_001)}
+      ]
+
+      assert_raise Postgrex.Error, fn ->
+        Collections.put_all(collection, items)
+      end
```

Comment:
I think we do want to do something about this at some point - unfortunately the Postgrex error isn't super detailed (or rather postgres isn't very specific). But lets add that to the backlog

Reply:
Yeah, not 100% satisfied but it was better than penalizing all calls to the function but checking item by item if they have valid length.